api/collection.go

```diff
@@ -78,6 +79,23 @@ func (api *API) AddCollectionHandler(w http.ResponseWriter, r *http.Request) {
 	log.Event(ctx, "add collection request completed successfully", log.INFO, logData)
 }
 
+func (api *API) validateCollection(ctx context.Context, collection *models.Collection) error {
+
```

Comment:
Can collection be nil?

Reply:
I don't think so - only if it successfully unmarshalls the collection will it call validate. I have added a check anyway to safeguard it.